R/aggregate_game_stats_kicking.R

```diff
@@ -0,0 +1,186 @@
+#' Summarize Kicking Stats
+#'
+#' @description Build columns that aggregate kicking stats at the game level.
+#'
+#' @param pbp A Data frame of NFL play-by-play data typically loaded with
+#' [load_pbp()] or [build_nflfastR_pbp()].
+#' @param weekly If `TRUE`, returns week-by-week stats, otherwise, stats for
+#' the entire season.
+#'
+#' @examples
+#' \donttest{
+#' pbp <- nflreadr::load_pbp(2021)
+#' calculate_player_stats_kicking(pbp, weekly = TRUE)
+#' }
+#'
+#' @return a dataframe of kicking stats
+#' @seealso <https://nflreadr.nflverse.com/reference/load_player_stats.html> for the nflreadr function to download this from repo (`stat_type = "kicking"`)
+#' @export
+calculate_player_stats_kicking <- function(pbp, weekly = FALSE) {
+
+  # need newer version of nflreadr to use load_players
+  rlang::check_installed("nflreadr (>= 1.3.0)")
+
+  # First, creating a grouping variable object to toggle the weekly argument w/
+  grp_vars <- list("season", "week", "season_type", "player_id", "team")
+  if (!weekly) {
+    grp_vars <- setdiff(grp_vars, c("week", "season_type"))
+  }
```

Comment:
```suggestion
  if (!weekly) grp_vars <- list("player_id","team")
```

This makes the aggregate function more consistent with how [offense](https://github.com/nflverse/nflfastR/blob/master/R/aggregate_game_stats.R#L707) and [defense](https://github.com/nflverse/nflfastR/blob/master/R/aggregate_game_stats_def.R#L552) get summarised - which gives flexibility to pass in multiple seasons and get one row back per player-team combo